pkg/server/server.go

```diff
@@ -446,6 +477,18 @@ func (s *ProxyServer) Connect(stream agent.AgentService_ConnectServer) error {
 	if err != nil {
 		return err
 	}
+	if s.proxyStrategy == "destHost" {
```

Comment:
Should the proxy server with this proxy strategy reject agents without the corresponding header? I'd imagine cases where a subset of agents are missing the header would be very difficult to detect.

Reply:
I was not sure if users forget to run agents with the corresponding proxy strategy (i.e., not providing corresponding headers), should the server still take the agent.   
Yes, I agree, this behavior can make it difficult to bug. Let's assume that if the server is running with a certain proxy strategy, agents not providing the required header will be rejected.